src/DataGrid.tsx

```diff
@@ -491,19 +481,21 @@ function DataGrid<R, SR, K extends Key>(
   }
 
   function onGridFocus() {
-    if (!isCellWithinBounds(selectedPosition)) {
-      // Tabbing into the grid should initiate keyboard navigation
-      const initialPosition: SelectCellState = { idx: 0, rowIdx: 0, mode: 'SELECT' };
-      if (isCellWithinBounds(initialPosition)) {
-        setSelectedPosition(initialPosition);
-      }
-    } else {
-      // otherwise if we already have a selected cell, we should scroll back to it when focusing the grid
-      scrollToCell(selectedPosition);
+    if (isCellWithinBounds(selectedPosition)) return;
+    // Tabbing into the grid should initiate keyboard navigation
+    const initialPosition: SelectCellState = { idx: 0, rowIdx: 0, mode: 'SELECT' };
+    if (isCellWithinBounds(initialPosition)) {
+      setSelectedPosition(initialPosition);
     }
+    // otherwise browser automatically scrolls to the selected cell
```

Comment:
It works well most of the time but if the cell is behind the frozen cell then it is scrolled.